src/index.ts

```diff
@@ -16,6 +16,30 @@ export interface ToolkitOptions {
 }
 
 export class Toolkit {
+  /**
+   * Run an asynchronous function that accepts a toolkit as its argument, and fail if
+   * an error occurs.
+   *
+   * @param func - Async function to run
+   * @param [opts] - Options to pass to the toolkit
+   *
+   * @example This is generally used to run a `main` async function:
+   *
+   * ```js
+   * Toolkit.run(async tools => {
+   *   // Action code here.
+   * }, { event: 'push' })
+   * ```
+   */
+  public static async run (func: (tools: Toolkit) => Promise<unknown>, opts?: ToolkitOptions) {
+    const tools = new Toolkit(opts)
+
+    return func(tools).catch(err => {
+      tools.log.fatal(err)
```

Comment:
`tools.exit.failure` can log a message as it's argument:

https://github.com/JasonEtco/actions-toolkit/blob/07e0eca27fb246116882844edb94ef7a9391debc/src/exit.ts#L37

But now I realize that its not using the logger 😱 I'll get that fixed to use `tools.log.fatal` inside of `tools.exit.failure()`, but it means that we could do:

```suggestion
      tools.exit.failure(err)
```